src/v/transform/transfer_queue.h

```diff
@@ -30,7 +30,8 @@ concept MemoryMeasurable = requires(const T v) {
 constexpr size_t default_items_per_chunk = 128;
```

Comment:
> ss::semaphore::wait` with an abort_source has some
stack-use-after-return issue that we haven't yet been able to track
down.

is this discussed somewhere else?

Reply:
No @ballard26 had mentioned that he was seeing segfaults in release mode and we both have seen `stack-use-after-return` with `ss::semaphore::wait`. This commit removes the asan violation from my testing.

We both took a look and it seems hard to repro in a unit test and it's not obvious where the bug is.

Reply:
https://redpandadata.slack.com/archives/C02H58RN215/p1706638052358819?thread_ts=1706626810.559049&cid=C02H58RN215